api/user.js

```diff
@@ -0,0 +1,57 @@
+const sa = require('superagent').agent();
+
+const { api_key = '', api_username = 'system', api_url = 'http://localhost:3000' } = require('../secrets');
+
+const specialUsernames = ['discobot', 'system'];
+
+// get a user that's not discobot or somesuch
+const getArbitraryUser = (exports.getArbitraryUser = () =>
+  sa
+    .get('http://localhost:3000/admin/users/list/active.json')
+    .query({ api_key, api_username })
+    .then(({ body }) => {
+      const users = body.filter(user => !specialUsernames.includes(user.username));
+      return users[0];
+    }));
+
+const getCustomUserFields = (exports.getCustomUserFields = () =>
+  sa
+    .get('http://localhost:3000/admin/customize/user_fields.json')
+    .query({ api_key, api_username })
+    .then(({ body: { user_fields } }) => user_fields));
+
+const createCustomUserField = (exports.createCustomUserField = async user_field => {
+  console.assert(['text', 'confirm', 'dropdown'].includes(user_field.field_type));
+
+  let options_string;
```

Comment:
Unless we're passing the variable into the fields thing, let's keep with the idiomatic camel casing in JavaScript. The only reason I had the API key and username like that was to make it convenient to put into the field object, but adding those to a request should probably be pushed off into a utility somewhere so we don't have to write the same query string and body code everywhere.

Reply:
Ah, that's why I was confused. Sounds good.